### PR TITLE
Feature/referential integrity

### PR DESCRIPTION
**Pull request by TomHAnderson**
Added referential integrity to the entire database and changed all integer primary keys and foreign keys to bigInteger.

See #29

### CODE REVIEW

**Comment by TomHAnderson** [comment]
This PR excludes referential integrity for all the audit fields:  created_by, updated_by, deleted_by.  

I feel that this PR is a big step in the right direction for the application and careful consideration should be made concerning the audit fields.  A true auditing solution may be a better approach than having these audit fields exist in so many tables.  A good example of their unnecessity is `vh_role_permissions` because the auditing data just isn't necessary for this administrative data, IMHO.

It is my hope that, having provided a .skipper file, that you will take the time to learn how to create the foreign key relationships for the auditing fields if that is what you decide to do.

Your new ERD can be viewed here:  https://github.com/webreinvent/vaahcms/blob/8e5af6b184118aafa6c69531376debf4d990f9a1/erd.pdf

**Comment by TomHAnderson** [comment]
Per https://github.com/webreinvent/vaahcms/issues/29#issuecomment-890315033 this `migrate` and `migrate:reset` commands both work as expected.

**Comment by themodernpk** [comment]
@TomHAnderson will review it soon. Thanks

File: Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php [hunk 1]
```diff
@@ -0,0 +1,1440 @@
...
+        });
+        Schema::table('vh_user_roles', function (Blueprint $table) {
+            $table->bigInteger('vh_user_id')->nullable(true)->unsigned()->after('id')->change();
+        });
+        Schema::table('vh_users', function (Blueprint $table) {
+            $table->bigInteger('id')->autoIncrement()->unsigned()->change();
+        });
+        /* Reordering of existing columns is not supported by Laravel Eloquent.
+        Schema::table('vh_theme_blocks', function (Blueprint $table) {
+            $table->moveColumnAfter('vh_theme_id', 'id');
```
Comment (themodernpk) [inline Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php:143]: @TomHAnderson `vh_theme_id` is after `id`, what are you suggesting here?

File: Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php [hunk 2]
```diff
@@ -0,0 +1,1440 @@
...
+        });
+        */
+        
+        /* But you can create your own migrations with following SQL query based on your target SQL engine.
+        DB::transaction(function () {
+            DB::statement('ALTER TABLE `vh_theme_blocks` MODIFY `vh_theme_id` [COLUMN_DEF] AFTER `id`');
+        });
+        */
+        /* Reordering of existing columns is not supported by Laravel Eloquent.
```
Comment (themodernpk) [inline Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php:152]: @TomHAnderson do I need to review your comments?

**Comment by themodernpk** [comment]
@TomHAnderson `vaahcms.skipper` please remove this file, I don't think that's required.

File: Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php (hunk 1 above)
Comment (TomHAnderson) [reply Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php:143]: I'll explain this as part of the removing .skipper file

File: Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php (hunk 2 above)
Comment (TomHAnderson) [reply Database/Migrations/2021_07_31_161559_skipper_migrations_2021073116155945.php:152]: No.  I'll explain this as part of removing the .skipper file

**Comment by TomHAnderson** [comment]
An explanation of how I created this migration file is in order.  

I used the ORM and Laravel-specific tool [Skipper](https://skipper18.com) to reverse engineer your database into an ERD.  I then added all the relationships between the tables and [arranged them](https://github.com/webreinvent/vaahcms/blob/572ad35cd858f3c5c96e8e4fe883687a9cf10a28/erd.pdf) into an organized Entity Relationship Diagram.  

I then exported the diagram into a [PDF](https://github.com/webreinvent/vaahcms/blob/572ad35cd858f3c5c96e8e4fe883687a9cf10a28/erd.pdf) and I exported the diagram into a Laravel migration.  **All of the code and comments in the migration were generated by Skipper.**

You may be thinking that I'm trying to tie your project to this incredible tool.  Nothing I've done has created such a connection.  However, I ask that you take the due diligence to explore what Skipper can bring to this database project.  Please start by reviewing what an ERD is by reading all of the preview lectures here:  https://erd.design/p/designing-entity-relationship-diagrams-in-laravel

Next, download a trial version of [Skipper](https://skipper18.com) and open the [vaahcms.skipper](https://raw.githubusercontent.com/webreinvent/vaahcms/572ad35cd858f3c5c96e8e4fe883687a9cf10a28/vaahcms.skipper) file included with this PR and explore what Skipper will do for your project.

It's your choice to reject Skipper and request the vaahcms.skipper file be removed from this PR but if that is done then the [erd.pdf](https://github.com/webreinvent/vaahcms/blob/572ad35cd858f3c5c96e8e4fe883687a9cf10a28/erd.pdf) file must also be removed because it is just a snapshot of the actual ERD (which is vaahcms.skipper).

In closing, I'm asking you to consider using Skipper for your ORM (and database) management in Laravel.  It is a unique application and the only database tool I'm aware of which can export to migrations (and models).

**Comment by themodernpk** [comment]
@TomHAnderson Most of your suggestions are accepted and released with `minor` version at:

[v1.5.0](https://github.com/webreinvent/vaahcms/releases/tag/v1.5.0 )

We may release a `patch` for lower versions later.

Unfortunately, we have decided not to use `skipper18` at the moment, instead, we choose `https://www.dbml.org/home/` to create visual & sharable database documentation. Feel free to review:

- [Database Table Structure](https://dbdocs.io/webreinvent/vaahcms)
- [Database Relationship](https://dbdocs.io/webreinvent/vaahcms?view=relationships)

Thank you so much for your suggestion. Do let us know if you have more suggestions.

Feel free join us at: [Slack](https://join.slack.com/t/vaah/shared_invite/zt-tlop66za-yKKtiuMZ5S4lAYUQ6gtfMw) and see roadmap at: [Trello](https://trello.com/b/MI18Rbs5/vaahcms)

**Pull request closed without merging**